Grid/qcd/utils/WilsonLoops.h

```diff
@@ -177,25 +177,37 @@ template <class Gimpl> class WilsonLoops : public Gimpl {
   }
 
   //////////////////////////////////////////////////
-  // average over all x,y,z the temporal loop
+  // average Polyakov loop in any direction (spatial or temporal)
   //////////////////////////////////////////////////
-  static ComplexD avgPolyakovLoop(const GaugeField &Umu) {  //assume Nd=4
+  static ComplexD avgPolyakovLoop(const GaugeField &Umu, const int mu) {  //assume Nd=4
+    
+    // TODO: Protect against mu<0 || mu>=4
     GaugeMat Ut(Umu.Grid()), P(Umu.Grid());
     ComplexD out;
     int T = Umu.Grid()->GlobalDimensions()[3];
     int X = Umu.Grid()->GlobalDimensions()[0];
     int Y = Umu.Grid()->GlobalDimensions()[1];
     int Z = Umu.Grid()->GlobalDimensions()[2];
 
-    Ut = peekLorentz(Umu,3); //Select temporal direction
+    // Number of sites in mu direction
+    int L = Umu.Grid()->GlobalDimensions()[mu];
```

Comment:
Usually I would interpret `L` as the spatial lattice volume; here it could also be temporal. Could `N_mu` be a good name here to represent the number of sites in direction $\mu$?